x/ibc/02-client/handler.go

```diff
@@ -27,6 +28,8 @@ func HandleMsgCreateClient(ctx sdk.Context, k Keeper, msg exported.MsgCreateClie
 		if err != nil {
 			return nil, err
 		}
+	case exported.Localhost:
+		clientState = localhost.NewClientState(ctx.MultiStore().GetKVStore(k.GetStoreKey()))
```

Comment:
This shouldn't be passed in the entire KVStore. Rather pass in a prefix store unique to the local client so we don't read/write other client data

Reply:
addressed on #6018